### DISCUSSION TITLE
Accepting remote URL for configuration file

### DISCUSSION THREAD

**Discussion by marcaurele** [orhun/git-cliff#940] [Ideas]
Hi!

I wish we could provide a remote URL to the command in order to store somewhere a common template for many repositories for which we generate the release notes with `git-cliff`. This in order to ensure each time the latest template is used from a central location.

**Comment by orhun** [comment]
Hello! Can you give an example of how this would look like? I'm not sure if I got the idea.

**Reply by marcaurele** [reply]
As a team, we would like to have a common configuration for the template, and reuse this same configuration across multiple repositories. So instead of committing a _git-cliff_ configuration file in each of them, which will become a pain to keep in sync in all of them, and because it's neither easy to deploy a shared configuration in the `$HOME/git-cliff/cliff.toml` directory on all developer's computer, having a config pushed at a location that is accessible by all devs at a given url, would solve the problem. I would like to execute the command with a URL for the config location:

```console
git-cliff --config https://my.developer/git-cliff/cliff.toml ...
```

**Reply by orhun** [reply]
Hmm, I see. Thanks for the detailed explanation :)

That's seems more like an infrastructure problem more than a `git-cliff` problem to me. Is there anything that prevents you from running e.g. `wget` and downloading the configuration file before running `git-cliff`? Because that's effectively what we are going to do (with a library) with that `--config` or (more preferably `--remote-config`) option.

**Reply by marcaurele** [reply]
Well, we could but it would requires us to manage this extra pre step across all our repos (ok) and having to deal with the temporary file somewhere. I simply thought to propose it as it would simplify things from the user point of view (not the development one it requires to deal with all sort of errors / bad format and data injection).

Feel free to close it if you do not want to implement it, no pb.

Thank you for the great job on this tool!

**Reply by orhun** [reply]
Got it. Actually I just saw [`zellij`](https://zellij.dev) supporting something similar with the layout files and I think I like it.

e.g.

```bash
zellij -l https://raw.githubusercontent.com/orhun/dotfiles/refs/heads/master/zellij/.config/zellij/layouts/monitoring.kdl
```

My perspective on this changed a bit so I might actually go ahead and implement remote configs for git-cliff sometime next week :)

**Reply by orhun** [reply]
Implemented in #1083 as a separate flag

**Reply by marcaurele** [reply]
@orhun any plan to push a new release soon so that we can benefit from that change?

**Reply by orhun** [reply]
Hello @marcaurele, yup, I'm still trying to finish up a couple of things

**Reply by TribuneX** [reply]
Thanks for that feature! I guess the repo hosting that config file must be public in order for `git-cliff` to be able to fetch the file. Any hints or tricks available to also load such a file from a private repository?

**Reply by orhun** [reply]
Hey again, yes. IIRC you could share files from private repositories using a private hash in the URL... otherwise I'm not sure
